plugins/mc_text_generator.js

```diff
@@ -0,0 +1,711 @@
+                                cubes: []
+                            }
+                        }
+                        
+                        let offset = 0
+                        let textCube, layerCube
+                        textLength = 0
+    
+                        Undo.initEdit({group: textGroup, elements: [], outliner: true})
+                        textGroup = new Group('text_' + formData.input).init()
+
+                        for (const char of formData.input.toLowerCase()) {
+                            letterGroup = new Group('character_' + char).init()
+
+                            // Check for an invalid character
+                            if (!charMap.hasOwnProperty(char)) {
+                                console.log(char)
+                                continue
+                            }
+
+                            randNum = getRandomInt(8)
+
+                            // Generate cubes based on letter
+                            for (const cube of charMap[char].cubes) {
+                                textCube = new Cube({
+                                    name: "text_" + formData.input,
+                                    from: [cube[0] + offset, cube[1], cube[2]],
+                                    to: [cube[3] + offset, cube[4], cube[5]],
+                                    color: randNum
+                                }).addTo(letterGroup).init()
+
+                                textCube.flip(0, 2.0, true)
+
+                                // Generate layer if user checked the box
+                                if (formData.generateLayer == true && formData.depth == 0) {
+                                    layerCube = new Cube({
+                                        name: "text_" + formData.input + "_layer",
+                                        from: [cube[0] + 0.2 + offset, cube[1] - 0.2, cube[2] + 0.2],
+                                        to: [cube[3] + offset + 0.2, cube[4] - 0.2, cube[5] + 0.2]
+                                    }).addTo(letterGroup).init()
+
+                                    layerCube.flip(0, 2.0, true)
+                                }
+                            }
+
+                            offset += charMap[char].width + formData.letterSpace
+                            textLength += charMap[char].width + formData.letterSpace
+
+                            letterGroup.addTo(textGroup)
+                        }
+
+                        textGroup.children.forEach(group => {
+                            group.children.forEach(cube => {
+                                cube.moveVector(textLength / 2 - 4, 0)
+                            })
+                        })
+                        
+                        editBone()
+                        Canvas.updateView({groups: [textGroup, Group.selected], transform: true});
+                        Undo.finishEdit("Generated Text", {elements: selected});
```

Comment:
The `outliner` and `group` aspects are missing in this undo statement, causing the redo function to not work correctly.